airbyte-integrations/connectors/destination-bigquery/build.gradle

```diff
@@ -55,6 +59,75 @@ dependencies {
     implementation libs.s3
 }
 
+def connectionSpecificationSchemaFile = file(
+        "${buildDir}/generated/sources/connectionSpecification/${project.name}-connection-specification.json")
+
+tasks.register('extractConnectionSpecificationSchema') {
+
+    def inputFile = file("${projectDir}/src/main/resources/spec.json")
+    inputs.file inputFile
+
+    def outputFile = connectionSpecificationSchemaFile
+    outputs.file outputFile
+
+    doFirst {
+        // Read the input JSON file
+        def inputJson = new JsonSlurper().parseText(inputFile.text)
+        logger.info "Parsed JSON from input: ${inputFile}"
+        def properties = inputJson["connectionSpecification"]["properties"]
+        // underlying jsonschema2pojo library cannot generate ambigious oneOf classes.
+        // alter the oneOf property to pull the common required fields to top level (should always be one in spec case)
+        // and add it as a property, let others be additionalProperties to use JsonNode functions
+        properties.each { property ->
+            if (property.value.type == "object" && property.value.oneOf != null) {
+                def requiredFieldsList = property.value.oneOf.collect {it.required}
+                def commonRequiredFields = requiredFieldsList.inject {acc, list -> acc.intersect(list)}
+                commonRequiredFields.each {fieldname ->
+                    def requiredObjectFromOneOf = property.value.oneOf.first()['properties'][fieldname as String]
+                    property.value.properties = [(fieldname):requiredObjectFromOneOf]
```

Comment:
this could also be represented as an enum with the values of `const` field, worried it might become too rigid incase we decide to change it later.